timesketch/lib/analyzers/sigma_tagger.py

```diff
@@ -70,47 +74,123 @@ def run(self):
             self.sigma_config, {})
         tags_applied = {}
 
+        simple_counter = 0
+
         rules_path = os.path.join(os.path.dirname(__file__), self._RULES_PATH)
-        for rule_filename in os.listdir(rules_path):
-            tag_name, _ = rule_filename.rsplit('.')
-            tags_applied[tag_name] = 0
-            rule_file_path = os.path.join(rules_path, rule_filename)
-            rule_file_path = os.path.abspath(rule_file_path)
-            logging.info('[sigma] Reading rules from {0!s}'.format(
-                rule_file_path))
-            with open(rule_file_path, 'r') as rule_file:
-                rule_file_content = rule_file.read()
-                parser = sigma_collection.SigmaCollectionParser(
-                    rule_file_content, self.sigma_config, None)
-                try:
-                    results = parser.generate(sigma_backend)
-                except NotImplementedError as exception:
-                    logging.error(
-                        'Error generating rule in file {0:s}: {1!s}'.format(
-                            rule_file_path, exception))
-                    continue
-
-                for result in results:
-                    logging.info(
-                        '[sigma] Generated query {0:s}'.format(result))
-                    number_of_tagged_events = self.run_sigma_rule(
-                        result, tag_name)
-                    tags_applied[tag_name] += number_of_tagged_events
+
+        # The extension to search for
+        exten = '.yml'
+
+        for dirpath, dirnames, files in os.walk(rules_path):
+            for rule_filename in files:
+                if rule_filename.lower().endswith(exten):
+                    print(os.path.join(dirpath, rule_filename))
+                    if os.path.isdir(os.path.join(rules_path, rule_filename)): # if a sub dir is found, append it to be scanned for rules
+                        logging.error(
+                            'this is a directory, not a file, skipping: {0:s}'.format(
+                                rule_filename))
+                        continue
+
+                    tag_name, _ = rule_filename.rsplit('.')
+                    tags_applied[tag_name] = 0
+                    rule_file_path = os.path.join(dirpath, rule_filename)
+                    rule_file_path = os.path.abspath(rule_file_path)
+                    logging.info('[sigma] Reading rules from {0!s}'.format(
+                        rule_file_path))
+                    with open(rule_file_path, 'r') as rule_file:
+                        try:
+                            rule_file_content = rule_file.read()
+                        except UnicodeDecodeError as exception:
```

Comment:
Why do we get a UnicodeDecodeError here? Shouldn't we support unicode files?

Reply:
true, removing it